src/test/java/core/basesyntax/handler/impl/BalanceOperationHandlerTest.java

```diff
@@ -0,0 +1,49 @@
+package core.basesyntax.handler.impl;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.handler.ShopOperationHandler;
+import core.basesyntax.model.FruitTransaction;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class BalanceOperationHandlerTest {
+    private static ShopOperationHandler balanceOperationHandler;
+    private FruitTransaction fruitTransaction;
+
+    @BeforeAll
+    static void beforeAll() {
+        balanceOperationHandler = new BalanceOperationHandler();
+    }
+
+    @BeforeEach
+    void setUp() {
+        fruitTransaction = new FruitTransaction();
+        fruitTransaction.setFruit("dummy");
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.fruitStorage.remove(fruitTransaction.getFruit());
+    }
+
+    @Test
+    void handle_validQuantityData_Ok() {
+        assertAll(() -> {
+                    fruitTransaction.setQuantity(0);
+                    balanceOperationHandler.handle(fruitTransaction);
+                }, () -> {
+                    fruitTransaction.setQuantity(12);
+                    balanceOperationHandler.handle(fruitTransaction);
+                }
```

Comment:
why set 0 before setting 12? how this benefits the test?

Reply:
> 
I check edge value (in this case "0"), but checking of quantity value wich was saved in Storage  is checked only  for 
fruitTransaction with quantity 12

Reply:
Done. Changed logic a bit.